nixos/lib/test-driver/test-driver.py

```diff
@@ -566,15 +492,12 @@ def screenshot(self, filename: str) -> None:
             filename = os.path.join(out_dir, "{}.png".format(filename))
         tmp = "{}.ppm".format(filename)
 
-        with self.nested(
-            "making screenshot {}".format(filename),
-            {"image": os.path.basename(filename)},
-        ):
-            self.send_monitor_command("screendump {}".format(tmp))
-            ret = subprocess.run("pnmtopng {} > {}".format(tmp, filename), shell=True)
-            os.unlink(tmp)
-            if ret.returncode != 0:
-                raise Exception("Cannot convert screenshot")
+        self.logger.info(f"making screenshot {filename}")
+        self.send_monitor_command("screendump {}".format(tmp))
```

Comment:
i know that you basically just changed the indentation of this code after removing the nesting that we don't support any longer. but can you please at the same time homogenize a bit the formatting to use more `f"screendump {tmp}"` style?

Reply:
Alternatively, `logging` has its own formatting which might make sense to use if we're committing to using it (`self.logger.info("making screenshot %s", filename)`)

Reply:
I am generally not against this, but the automatic type selection using `{}` looks favorable to me in contrast to `%s` style formatting.

Reply:
I believe `%s` will always act the same as `{}`:
```
>>> "%s" % 5.6
'5.6'
>>>
```

(unlike C where it causes horrifying undefined behaviour :D)

Reply:
No, the PR is already quite big. This is another refactoring.